utilities/resource_sync/sync.py

```diff
@@ -93,21 +111,17 @@
 if args.dst_region is not None:
     dst_session_args['region_name'] = args.dst_region
     logger.info(f"Destination: boto3 Session uses {args.dst_region} region based on the argument.")
-
-src_session = boto3.Session(**src_session_args)
-src_glue = src_session.client('glue', endpoint_url=args.src_glue_endpoint_url)
-src_s3 = src_session.resource('s3', endpoint_url=args.src_s3_endpoint_url)
-
-if args.sts_role_arn is not None:
-    src_sts = src_session.client('sts', endpoint_url=args.src_sts_endpoint_url)
-    res = src_sts.assume_role(RoleArn=args.sts_role_arn, RoleSessionName='glue-job-sync')
+if args.dst_role_arn is not None or args.sts_role_arn is not None:
+    dst_sts = boto3.client('sts', endpoint_url=args.dst_sts_endpoint_url)
+    role_arn = args.dst_role_arn if args.dst_role_arn is not None else args.sts_role_arn
+    res = dst_sts.assume_role(RoleArn=role_arn, RoleSessionName='glue-job-sync-dst')
     dst_session_args['aws_access_key_id'] = res['Credentials']['AccessKeyId']
     dst_session_args['aws_secret_access_key'] = res['Credentials']['SecretAccessKey']
     dst_session_args['aws_session_token'] = res['Credentials']['SessionToken']
 
-if args.dst_profile is None and args.sts_role_arn is None:
```

Comment:
Do we want to remove this validation "if all of args.dst_profile/sts_role_arn/dst_role_arn"?

Reply:
oops I missed that validation. nevermind.